src/dnn/descriptors.jl

```diff
@@ -0,0 +1,91 @@
+const MIOPEN_DIM_MAX = 5
+
+function check_ndims(x)
+    if ndims(x) > MIOPEN_DIM_MAX
+        error(
+            "MIOpen supports only up to `$MIOPEN_DIM_MAX` dimensions.\n" *
+            "Tensor has `$(ndims(x))` dimensions.")
+    end
+end
+
+mutable struct TensorDescriptor
+    handle::miopenTensorDescriptor_t
+    dtype::miopenDataType_t
+end
+
+function TensorDescriptor(
+    dtype::miopenDataType_t, dims::Int64,
+    sizes::Vector{Int32}, strides::Vector{Int32},
+)
+    handle_ref = Ref{miopenTensorDescriptor_t}()
+    miopenCreateTensorDescriptor(handle_ref) |> check
+
+    handle = handle_ref[]
+    miopenSetTensorDescriptor(handle, dtype, dims, sizes, strides) |> check
+    d = TensorDescriptor(handle, dtype)
+
+    finalizer(d) do d_
+        miopenDestroyTensorDescriptor(d_.handle) |> check
+    end
+    d
+end
+
+function TensorDescriptor(x::ROCArray{T}) where T
+    check_ndims(x)
+    dtype = get_miopen_data_type(T)
+    # NOTE: Dims and strides are reversed to support WHCN convolutions.
+    sizes = Int32[reverse(size(x))...]
+    strides_ = Int32[reverse(strides(x))...]
+    TensorDescriptor(dtype, ndims(x), sizes, strides_)
+end
+
+function Base.ndims(desc::TensorDescriptor)
+    nd = Ref{Int32}(0)
+    miopenGetTensorDescriptorSize(desc.handle, nd) |> check
+    Int64(nd[])
+end
+
+function get(desc::TensorDescriptor)
```

Comment:
Can we rename this to something like `unpack` or `params` or similar? `get` is somewhat ambiguous in what it actually gets.